src/ansitest/plugin.py

```diff
@@ -0,0 +1,508 @@
+    cd_tox_dir = f"cd {TOX_WORK_DIR}"
+    rsync_cmd = f'rsync -r --cvs-exclude --filter=":- .gitignore" . {galaxy_build_dir}'
+    full_cmd = f"bash -c '{cd_tox_dir} && {rsync_cmd}'"
+    commands.append(full_cmd)
+    if in_action():
+        commands.append(end_group)
+
+    if in_action():
+        group = "echo ::group::Build and install the collection"
+        commands.append(group)
+    cd_build_dir = f"cd {galaxy_build_dir}"
+    build_cmd = "ansible-galaxy collection build"
+    tar_file = f"{c_namespace}-{c_name}-*.tar.gz"
+    install_cmd = f"ansible-galaxy collection install {tar_file} -p {collections_root}"
+    full_cmd = f"bash -c '{cd_build_dir} && {build_cmd} && {install_cmd}'"
+    commands.append(full_cmd)
+    if in_action():
+        commands.append(end_group)
+
+    if in_action():
+        group = "echo ::group::Initialize the collection to avoid ansible #68499"
+        commands.append(group)
+    cd_install_dir = f"cd {collection_installed_at}"
+    git_cfg = "git config --global init.defaultBranch main"
+    git_init = "git init ."
+    full_cmd = f"bash -c '{cd_install_dir} && {git_cfg} && {git_init}'"
+    commands.append(full_cmd)
+    if in_action():
+        commands.append(end_group)
+
+    if env_conf.name == "sanity-py3.8-2.9":
+        # Avoid "Setuptools is replacing distutils"
+        if in_action():
+            group = "echo ::group::Use old setuptools for sanity-py3.8-2.9"
+            commands.append(group)
+        pip_install = "pip install setuptools==57.5.0"
+        commands.append(pip_install)
+        if in_action():
+            commands.append(end_group)
+
+    return commands
+
+
+def conf_deps(env_conf: EnvConfigSet, test_type: str) -> str:
+    """Add dependencies to the tox environment.
+
+    :param env_conf: The environment configuration.
+    :param test_type: The test type.
+    :return: The dependencies.
+    """
+    deps = []
+    if test_type in ["integration", "unit"]:
+        deps.extend(OUR_DEPS)
+        try:
+            with (TOX_WORK_DIR / "test-requirements.txt").open() as fileh:
+                deps.extend(fileh.read().splitlines())
+        except FileNotFoundError:
+            pass
+        try:
+            with (TOX_WORK_DIR / "requirements.txt").open() as fileh:
+                deps.extend(fileh.read().splitlines())
+        except FileNotFoundError:
+            pass
+
+    ansible_version = env_conf.name.split("-")[2]
+    base_url = "https://github.com/ansible/ansible/archive/"
+    if ansible_version in ["devel", "milestone"]:
+        ansible_package = f"{base_url}{ansible_version}.tar.gz"
+    else:
+        ansible_package = f"{base_url}stable-{ansible_version}.tar.gz"
+    deps.append(ansible_package)
+    return "\n".join(deps)
+
+
+def conf_passenv() -> List[str]:
+    """Build the pass environment variables for the tox environment.
+
+    :return: The pass environment variables.
+    """
+    passenv = []
+    passenv.append("GITHUB_TOKEN")
```

Comment:
Funny I was thinking about this over the weekend, we should definately allow the user to provide some customization of the environment variables.

I was thinking they could go in the ansible section of the tox.ini so they would not interfere with other environments...

Reply:
So it would probably be even better if this could be configured on a per-env basis. I wonder if, instead of injecting this MemoryLoader with `.insert(0, loader)` you could do `.append(loader)` and have the pre-programmed values in the plugin be the defaults for Ansible test cases while allowing them to be overridden in the tox.ini file?

Like, could I have

```ini
[testenv:unit-py3.11-2.14]
passenv = MYENV_*
```

and accept the rest of the values from this plugin but just override that? In my local testing that is exactly how it works. When I change the insert to an append, it allows the user to configure the test case however they want in the tox.ini but still accept my defaults from a plugin. Then we can have very conservative defaults, like this one, and allow the user to override it if they are running in Jenkins, CircleCI, etc.

Reply:
If I recall, the reason I went down the path of a new MemoryLoader was, if I recall, there was some attributes I couldn't modify. I can try it again.

For now, I've extracted the values from the ini loader and added them to the custom values in the MemoryLoader, it isn't great.

Reply:
No, that's not quite what I mean. I'll comment on the line that I mean, along with the examples I ran to validate what Tox is doing internally.

Reply:
Got it, I mis understood, agreed, let it work like native tox.  `append` is much cleaner

Reply:
I did find one thing interesting switching to append from insert. Any configuration in `[testenv]` overrides the configuration injectect by the plugin. I would have expected the environment specific configuration provided by the plugin to override anything in the root ini configuration.

I added some explanation to the README

Reply:
Hmmm, if you think that's undesirable we could maybe have it insert to the -1 position of the list? But then you're really relying on specific internals of tox.ini to have its loaders all in a specific order. In my opinion, the behavior you mentioned is fine, as it is still explicit configuration in `[testenv]` overriding implicit configuration. How do you feel on it?

Reply:
I think it's fine, although unexpected.  If someone want to modify the tox-ansible provided configuration, they have a few choices, redefine the provided in the custom, raise a PR/issue to modifu the default, or not use the plugin at all.

The only thing that leaves me a little worried is that I had envisioned this being a plugin that would work along side any exisitng tox.ini file without modification, and be enabled with the `--ansible` flag. Thinking about it now, the use could specific a different tox.ini file for ansible related activity or we could possibly inject the alternate for them (basically change the default from tox.ini to tox-ansible.ini.  I'll do some testing here.

Reply:
The plugin now suggests the use of `tox-ansible.ini` instead of `tox.ini` and will warn if the user is using the later. I think it is cleaner (and safer) and allows the plugin to be adopted without modifications of an existing ini file that might be in place.

LMK what you think